applications/FluidDynamicsApplication/custom_elements/embedded_navier_stokes.h

```diff
@@ -534,17 +434,22 @@ class EmbeddedNavierStokes : public NavierStokes<TDim, TNumNodes>
     ///@name Protected Operations
     ///@{
 
-    unsigned int ComputeSplitting(const ElementDataType& data,
-                                  MatrixType& rShapeFunctionValues,
-                                  VectorType& rGaussVolumes,
-                                  VectorType& rPartitionSigns,
-                                  const array_1d<double,TNumNodes>& distances,
-                                  VectorType& rEdgeAreas)
+    /**
+    * Computes the element splitting according to the distance values.
+    * @param rData: reference to element data structure
+    * @param rDistances: reference to an array containing the nodal distance values
+    * @param rSplittingData: reference to the intersection data structure
+    */
+    void ComputeSplitting(const ElementDataType& rData,
+                          const array_1d<double,TNumNodes>& rDistances,
+                          ElementSplittingDataStruct& rSplittingData)
     {
 
-        unsigned int ndivisions = 1;
-        VectorType NodalDistances = distances;
-        MatrixType NoSplitGradients = data.DN_DX;
+        rSplittingData.edge_areas = ZeroVector((TDim-1)*3);         // Should be initialized to the edge number
```

Comment:
@RiccardoRossi In fact, in here is where I allocate the memory. To use an array_1d would imply to modify the SplitTetrahedraUtilities() in the core, which is something that I think we do not want to do...